### PR TITLE
Enabling Shopify Build Pipeline: opencensus-service

### PR DESCRIPTION
**Pull request by shopify-admins**
## Enabling the Shopify Build Pipeline: opencensus-service

This PR adds Shopify Build to this repository.

**Please review this PR before merging it!**

- [ ] Head over to the Shopify Build [documentation](https://apidocs.shopify.io/shopify-build/) to learn more about configurating it.
- [ ] Examine your `shipit.yml` configuration file as it may be configured to depend on the [CircleCI status](https://github.com/Shopify/shipit-engine#ci). You can reference a pipeline in the `shipit.yml` file by using `buildkite/opencensus-service` as seen [here](https://github.com/Shopify/shopify-build/blob/f263a410151e4e56ee610801cfbc285e0b69f25d/shipit.yml#L19).
- [ ] Don't forget to replace your old CI badge with the Shopify Build one. You can get it [here](https://buildkite.com/shopify/opencensus-service/settings/badges).
- [ ] If you are using codecov.io for code coverage, you'll need to add the `CODECOV_TOKEN` environment variable to your Shopify Build pipeline. The token can be found at [codecov](https://codecov.io/gh/Shopify/opencensus-service/settings) and should be placed in the environment section of the `.shopify-build/secrets.ejson` file. [More info](https://apidocs.shopify.io/shopify-build/jobs/defining_jobs/#secrets-as-environment-variables).
- [ ] Rebase and squash the commits into one and don't forget to remove the `[ci skip]` part from the commit message:
      `git rebase -i 73558e6c42b145a03d28d1f712d9cfd1d06438db`
- [ ] We disabled commit statuses on new PRs for your pipeline to prevent other branches from being blocked while you are setting up Shopify Build. [See the builds for your opencensus-service pipeline](https://buildkite.com/shopify/opencensus-service/builds?branch=shopify-build-add-opencensus-service).

## Cleanup After Merging

- [ ] Enable commit statuses for the PRs in your repository by running (You will have to rerun your builds to start seeing them):
      `spy build status yes opencensus-service`.
- [ ] After merging the changes to set up Shopify Build you can remove CircleCI from your project by running `spy circle remove Shopify/opencensus-service`.
- [ ] Let people know with branches that are behind the `master` branch that their PRs are failing, because they are missing the necessary configuration files. You can use the following reply: “We migrated this repository from CircleCI to Shopify Build. Your build is failing, because your branch doesn't include the necessary configuration files. The configuration files are missing because your branch is behind the `master` branch. You should rebase your branch to make the build pass with `git pull --rebase origin master`”

## Questions about this PR?
Join the `#shopify-build` Slack channel.

Annoyed by how much you have to do yourself? Make it better in [spy](https://github.com/Shopify/spy/blob/master/app/shopify_build.rb).

### CODE REVIEW

File: .shopify-build/opencensus-service.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    docker: golang:1.12
+    workdir: /opencensus-service
+
+steps:
+- build: production
+  force: true
+- run: apt-get update && apt-get install time && make install-tools && make travis-ci
+  label: tests
+  container: testing
+  timeout: 30m
```
Comment (ibawt) [inline .shopify-build/opencensus-service.yml]: TODO: reduce this

File: .shopify-build/opencensus-service.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    docker: golang:1.12
+    build:
+      command: pipa build -x --push -- ./build.sh
+  testing:
+    docker: golang:1.12
+    workdir: /opencensus-service
+
+steps:
+- build: production
+  force: true
```
Comment (ibawt) [inline .shopify-build/opencensus-service.yml:15]: see if we need this still

File: Makefile [hunk 3]
```diff
@@ -6,7 +6,7 @@ ALL_SRC := $(shell find . -name '*.go' \
 # ALL_PKGS is used with 'go cover'
 ALL_PKGS := $(shell go list $(sort $(dir $(ALL_SRC))))
 
-GOTEST_OPT?=-v -race -timeout 30s
+GOTEST_OPT?=-v -race -timeout 60s
```
Comment (ibawt) [inline Makefile]: remove this

File: Makefile [hunk 4]
```diff
@@ -38,7 +38,7 @@ test:
 	$(GOTEST) $(GOTEST_OPT) $(ALL_PKGS)
 
 .PHONY: travis-ci
-travis-ci: fmt install-go-cmp vet lint test-with-cover
+travis-ci: fmt vet lint test-with-cover
```
Comment (ibawt) [inline Makefile:41]: this is from head

**Comment by ibawt** [comment]
@fbogsany feel free to review now.  
## Things
- rebased with master
- required force to build the container, I'll follow up with shopify-build but not a huge deal
- skipped a few "flakey" tests, I imagine there's others and will continue to be a problem. 
- it's using a custom dockerfile to facilitate our build system, they aren't doing anything exciting
   - we should probably not use `FROM scratch`, it makes debugging really hard. though not impossible
   - lets do that when I do the "proper" receiver

**Review by fbogsany — COMMENTED** [review]
LGTM

**Pull request merged**